Add prettier as a pre-commit hook

njerig opened the pull request:
The hook setup script is `setup-hooks.sh` and the hooks folder is `.githooks`.

To set up (only required once for a local repo):
1. `chmod +x ./setup-hooks.sh`
2. `./setup-hooks.sh`

This symlinks all hooks in `.githooks` to `.git/hooks` in your local repo.

njerig committed:
Fix case where there are no files to commit

njerig committed:
Fix spacing

njerig committed:
Add pre-commit setup note

njerig committed:
Tweaks

bvngee commented:
Looks great! couple questions:
\- in order for `npx prettier ...` to work we need to install prettier as a devDependency to the frontend (right now we just assume the user has it installed). Can you do that for this PR (just run `npm install --save-dev --save-exact prettier` inside of frontend/)?
\- Also I'm not sure but does the pre-commit script need to be updated to run prettier inside of frontend/ only or does it also work in the root? my guess is it probably still works
\- obviously won't work on windows. maybe its worth writing powershell/batch equivalent scripts too if we're going through the effort?

njerig committed:
Add prettier as dev package

njerig committed:
Spacing tweaks

njerig committed:
Add Windows support for pre-commit hook

njerig commented:
1. Added prettier as dev dependency in [ba06e95](https://github.com/formulaslug/telemetry-vis-software/pull/4/commits/ba06e95a38c016fb90c35785f38033e36323446f)
2. It works in root as well.
3. Just added Windows support [here](https://github.com/formulaslug/telemetry-vis-software/pull/4/commits/c4b646d94d8f1b249b10f4603c73baefb8dd8549). It uses PowerShell if the user doesn't have git bash (which comes with git) installed.

bvngee committed:
Discard changes to frontend/package.json

bvngee committed:
Discard changes to frontend/package-lock.json

bvngee committed:
re-add prettier

bvngee commented:
Tested, this seems to work! Thanks
